hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -836,7 +836,9 @@ private EntityDomainType resolveEntityReference(String entityName) {
 		log.debugf( "Attempting to resolve path [%s] as entity reference...", entityName );
 		EntityDomainType reference = null;
 		try {
-			entityName = creationContext.getJpaMetamodel().qualifyImportableName( entityName );
+			if ( !creationOptions.useStrictJpaCompliance() ) {
```

Comment:
The intention was more to verify that a FQN was not used.  While this change does protect that from happening, it does not (probably) give a very help exception

Reply:
thanks. This is a draft PR for I have not enough time for this. Definitively will spend more efforts.

Reply:
The "correct" thing to do would be to add this check after we have `reference`.  From there, if compliance is requested, we would check its `javax.persistence.metamodel.EntityType#getName` (as opposed to `org.hibernate.metamodel.model.domain.EntityDomainType#getHibernateEntityName`) to make sure it matches the incoming `entityName`.  HTH